docs/design.md

```diff
@@ -0,0 +1,273 @@
+
+As the managed resource of Ansible provider, once the `AnsibleRun` resource is defined, the provider will download the Ansible contents into the provider working directory first via the method `Connect()` in AnsibleRun controller, then trigger the Ansible run according to the provider lifecycle via method `Observe()/Create()/Update()/Delete()`. 
+
+Inside the provider, it actually relies on the command line tool [`ansible-galaxy`](https://docs.ansible.com/ansible/latest/cli/ansible-galaxy.html) to download Ansible contents, then use [`ansible-runner`](https://ansible-runner.readthedocs.io/) to execute them. The module [`ansible-runner-http`](https://github.com/ansible/ansible-runner-http) is a plugin for ansible runner that allows to emit Ansible status and events to HTTP services in the form of POST events, so as to notify AnsibleRun controller about the run results and attatch that to `AnsibleRun` as status.
+
+### Working Directory
+
+The provider working directory is used to host the Ansible contents downloaded from remote place. It is currently inside the provider container so that will not be persisted permanently by the provider. As a result, when the provider pod restarts, the contents will be lost, but the provider will download them from remote place again.
+
+## Supported Sources
+
+There are two types of sources from which the Ansible contents can be retrieved, installed and run by Crossplane Provider for Ansible (or Ansible provider for short).
+
+### Inline
+
+This is maily for quick test. You can inline the Ansible content to be run in an `AnsibleRun` resource. It will be wrapped as a `playbook.yml` file and stored in the working directory for the provider to run.
+
+Here is an example to call a builtin Ansible module:
+
+```yaml
+apiVersion: ansible.crossplane.io/v1alpha1
+kind: AnsibleRun
+metadata:
+  name: inline-example
+spec:
+  forProvider:
+    # For simple cases, you can use an inline source to specify the content of
+    # playbook.yml as opaque and inline yaml.
+    playbookInline: |
+      ---
+      - hosts: localhost
+        tasks:
+          - name: simple-playbook
+            debug:
+              msg: You are running 'simple-playbook'
+  providerConfigRef:
+    name: provider-config-example
+```
+
+### Remote
+
+This is more useful for a real project where Ansible contents are hosted in a remote place. The Ansible contents can be retrieved from [Ansible Galaxy](https://galaxy.ansible.com/) as community contents, or Automation Hub as Red Hat certified and supported contents, or a private Automation Hub that hosts private contents created and curated by an organization, or even a GitHub repository.
+
+Here is an example to run one Ansible role. If multiple roles are listed, they will be run sequencially one after another. 
```

Comment:
can you mention that this is a specific case for a `collection` type role

Reply:
ok